includes/Traits/URL_Aware.php

```diff
@@ -154,7 +154,27 @@ protected function go_to( $url ) {
 		$GLOBALS['wp']->public_query_vars  = $public_query_vars;
 		$GLOBALS['wp']->private_query_vars = $private_query_vars;
 
-		_cleanup_query_vars();
+		// Clean up query vars.
+		foreach ( $GLOBALS['wp']->public_query_vars as $v ) {
+			unset( $GLOBALS[ $v ] );
+		}
+
+		foreach ( $GLOBALS['wp']->private_query_vars as $v ) {
+			unset( $GLOBALS[ $v ] );
+		}
+
+		// Set up query vars for taxonomies and post types.
+		foreach ( get_taxonomies( array(), 'objects' ) as $t ) {
+			if ( $t->publicly_queryable && ! empty( $t->query_var ) ) {
+				$GLOBALS['wp']->add_query_var( $t->query_var );
+			}
+		}
+
+		foreach ( get_post_types( array(), 'objects' ) as $t ) {
+			if ( is_post_type_viewable( $t ) && ! empty( $t->query_var ) ) {
+				$GLOBALS['wp']->add_query_var( $t->query_var );
+			}
+		}
```

Comment:
This change seems unrelated. Can you explain what this does and why it's needed? Is there any overlap with the main change here?

Reply:
I was experiencing errors when the `_cleanup_query_vars()` function call was used when testing manually. From what I can see the `_cleanup_query_vars()` is a test utility class which explains why this was working in unit tests. The contents here matches that of the function (and poc).

Reply:
@jjgrainger Good catch, thanks!